onionshare/onionkey.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+"""
+OnionShare | https://onionshare.org/
+
+Copyright (C) 2017 Micah Lee <micah@micahflee.com>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import base64
+import hashlib
+import os
+
+import nacl.signing
+
+from Crypto.PublicKey import RSA
+
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.asymmetric import rsa
+
+
+b = 256
+
+def bit(h, i):
+    return (h[i // 8] >> (i % 8)) & 1
+
+
+def encodeint(y):
+    bits = [(y >> i) & 1 for i in range(b)]
+    return b''.join([bytes([(sum([bits[i * 8 + j] << j for j in range(8)]))]) for i in range(b // 8)])
+
+
+def H(m):
+    return hashlib.sha512(m).digest()
+
+
+def expandSK(sk):
```

Comment:
I think it would be beneficial for readability to collect all functions related to base64 blob generation under a single function (I later edited the code used here -- [here's the current version](https://github.com/micahflee/onionshare/issues/461#issuecomment-360971386)).

Reply:
Thanks @maqp - so this is an abandoned PR, the new version is at https://github.com/micahflee/onionshare/pull/731 . But I'll try and factor in these changes there.

Reply:
Ah, very well. Thank you!